ironfish/src/network/peers/peer.ts

```diff
@@ -439,22 +444,8 @@ export class Peer {
    * @param address Hostname of the address, or null to remove the address.
    * @param port Port to connect over. Must be null if address is null.
    */
-  setWebSocketAddress(address: string | null, port: number | null): void {
-    if (address === null && port !== null) {
-      throw new Error(
-        `Called setWebSocketAddress on ${String(
-          this.state.identity,
-        )} with a port but no address`,
-      )
-    }
-
-    // Don't do anything if the address and port stay the same
-    if (address === this._address && port === this._port) {
-      return
-    }
-
-    this._address = address
-    this._port = port
+  setWebSocketAddress(wsAddress: WebSocketAddress | null): void {
```

Comment:
Still sorta thinking through this, but I wonder if it makes more sense to have this function signature stay how it was (`setWebSocketAddress(address: string | null, port: number | null)`).

It feels like we're creating a less-safe function by making it the responsibility of the function-caller to do the error check that this function was doing. By keeping the function signature the same, we encapsulate this a bit better: function callers don't have to care about the specifics, and internally we get to utilize this type

It would be nice if we could have a constructor on a type definition, so we could contain the logic there, but alas. Making it a full-fledged class seems like the wrong approach, as well.